django/thunderstore/social/providers.py

```diff
@@ -125,33 +152,55 @@ def get_user_email(self) -> str:
         If user has chosen to make their email hidden, it won't be
         included in the user's basic information, but it can be fetched
         with a separate request.
-
-        https://docs.github.com/en/rest/users/emails
         """
         token = self._raise_for_token(self.token)
         url = "https://api.github.com/user/emails"
-        emails = self._fetch_from_api(token, url)
-        primary = next((email for email in emails if email["primary"]), None)
 
-        if primary is None or not primary["verified"]:
+        class PartialEmail(BaseModel):
+            """
+            For full schema see https://docs.github.com/en/rest/users/emails
+            """
+
+            email: str
+            primary: bool
+            verified: bool
+
+        class EmailList(BaseModel):
+            __root__: List[PartialEmail]
+
+            def __iter__(self):
+                return iter(self.__root__)
+
+        emails = self._fetch_from_api(token, url, EmailList)
+        primary = next((email for email in emails if email.primary), None)
+
+        if primary is None or not primary.verified:
             raise Exception("User has no email available")
 
-        return primary["email"]
+        return primary.email
 
     def get_user_info(self) -> Dict:
         """
         Fetch user info from GitHub's API using the access token.
-
-        https://docs.github.com/en/rest/users/users
         """
         token = self._raise_for_token(self.token)
         url = "https://api.github.com/user"
-        data = self._fetch_from_api(token, url)
+
+        class PartialResponseSchema(BaseModel):
+            """
+            For full schema see https://docs.github.com/en/rest/users/users
+            """
+
+            email: Optional[str]
```

Comment:
Add the default as `None`

Reply:
Realized this while reviewing the next PR, but better late than never 😅

Reply:
AFAIK the default is for the cases where no value is provided to the validator. In this use case, value is always provided (by the API), but it just might be None. So the default is not strictly speaking required, but I added it anyway.

Also added validation/typing for the value returned by the `get_user_info` methods and updated the test cases to match.

Reply:
I'll mark this as resolved given what @anttimaki said is true, and in fact it's better to leave a default value out if we always expect one to exist in the response (even if null).